weather_mv/loader_pipeline/bq.py

```diff
@@ -252,9 +271,9 @@ def expand(self, paths):
             (
                     extracted_rows
                     | 'WriteToBigQuery' >> WriteToBigQuery(
-                        project=self.table.project,
-                        dataset=self.table.dataset_id,
-                        table=self.table.table_id,
+                        project=self.project,
+                        dataset=self.dataset_id,
+                        table=self.table_id,
```

Comment:
If we wanted to create the BQ table in the pipeline, a simpler solution would be to change this step's disposition: https://beam.apache.org/documentation/io/built-in/google-bigquery/#create-disposition

Reply:
On second thought, we may want to keep your step since we create an opinionated schema...

Reply:
@alxmrs 
i tried putting the create table as a transform in the pipeline but faced many issues
- I tried the `sample` transform as you suggested but there is no way to ensure table creation happens before the usual pipeline flow. (apache_beam python sdk doesn't have support for `Wait.on` which could have made this possible)
- I also tried stateful processing but as we are windowing out pub sub reads, and a state is discarded when the window is expired, the solution didn't work.

I think there is potential in using the `create-disposition` flag in `WriteToBigQuery`. Can you please elaborate on what do you mean by `opinionated schema`

cc: @mahrsee1997

Reply:
Now that you mention it, I agree that using the create disposition is easiest! We need to make sure that we pass in our computed `schema` (say, from the `init`) into this transform instead of having it automatically make the schema. Thinking it over now, that's what my concern was about: I wasn't sure if we'd get the schema we wanted if it computed it automatically.

Reply:
Was surfing web for some other issue & found this. It might be helping here.

https://beam.apache.org/releases/pydoc/current/apache_beam.io.gcp.bigquery.html#module-apache_beam.io.gcp.bigquery